app/main.py

```diff
@@ -1,25 +1,99 @@
+from __future__ import annotations
 from abc import ABC
 
 
 class IntegerRange:
-    pass
+    def __init__(
+            self,
+            min_amount: int,
+            max_amount: int
+    ) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(
+            self,
+            owner: SlideLimitationValidator,
+            name: str
+    ) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(
+            self,
+            instance: SlideLimitationValidator,
+            owner: SlideLimitationValidator
+    ) -> int:
+        return getattr(instance, self.protected_name)
+
+    def __set__(
+            self,
+            instance: SlideLimitationValidator,
+            value: int
+    ) -> None:
+        if not self.min_amount <= value <= self.max_amount:
+            setattr(instance, self.protected_name, False)
+            return
 
 
 class Visitor:
-   pass
+    def __init__(self, name: str, age: int, weight: int, height: int) -> None:
+        self.name = name
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class SlideLimitationValidator(ABC):
-    pass
+    def __init__(
+            self,
+            age: int,
+            weight: int,
+            height: int
+    ) -> None:
+        self.age = age
+        self.weight = weight
+        self.height = height
+
+    def checking(self) -> bool:
+        if False in self.__dict__.values():
+            return False
+        return True
 
 
-class ChildrenSlideLimitationValidator(SlideLimitationValidator):
-    pass
+class ChildrenSlideLimitationValidator(
+    SlideLimitationValidator
+):
+    age = IntegerRange(4, 14)
+    height = IntegerRange(80, 120)
+    weight = IntegerRange(20, 50)
 
+    def __init__(self, age: int, weight: int, height: int) -> None:
+        super().__init__(age, weight, height)
 
-class AdultSlideLimitationValidator(SlideLimitationValidator):
-    pass
+
+class AdultSlideLimitationValidator(
+    SlideLimitationValidator
+):
+    age = IntegerRange(14, 60)
+    height = IntegerRange(120, 220)
+    weight = IntegerRange(50, 120)
+
+    def __init__(self, age: int, weight: int, height: int) -> None:
+        super().__init__(age, weight, height)
 
 
 class Slide:
-    pass
+    def __init__(
+            self,
+            name: str,
+            limitation_class: SlideLimitationValidator
+    ) -> None:
+        self.name = name
+        self.limitation_class = limitation_class
+
+    def can_access(self, visitor: Visitor) -> Visitor | None:
+        return self.limitation_class(
+            visitor.age,
+            visitor.weight,
+            visitor.height
+        ).checking()
```

Comment:
In the `can_access` method, the return type is hinted as `Visitor | None`, but you are returning the result of the `checking` method which is a boolean. Consider revising your type hints or the return statement.